### DISCUSSION TITLE
Accessing URL parameters with RouterViewLayout

### DISCUSSION THREAD

**Discussion by loisvazquezind** [Kitware/trame#139] [Q&A] [answered]
Hello everyone.

I am trying to create an application that allows me to load different files based on a parameter of the access URL. My idea would be, for example: to access localhost:8080/index.html#id=1234, and dump that 'id' parameter into a Python variable to later read the 1234.vtu file.

The first step I need to do is to dump said 'id' into a Python variable. In [this example](https://github.com/Kitware/trame/blob/master/examples/validation/core/10_router.py#L84) I see that it can be read via the variable '$route.params. id'. However, I can't find a way to assign that value to a Python variable. Ideally, it would be optimal to be able to dump the value of $route.params into a dictionary to handle the URL parameters later.

Any idea how to do it?

Thank you very much in advance.

**Comment by jourdain** [comment, marked as answer]
This was answered [here](https://github.com/Kitware/trame/discussions/83)

But if you really want to use the router something similar can still be done...

**Reply by loisvazquezind** [reply]
Excuse me Mr. Jourdain. I had not interpreted that discussion as related to my problem. Thank you very much for the prompt response, I have solved the problem.
